src/ol-layerswitcher.js

```diff
@@ -109,21 +109,48 @@ export default class LayerSwitcher extends Control {
         }
     }
 
+    /**
+    * Dispatch panel render events.
+    */
+    panelEvent(render_event) {
+        var retVal;
+        switch(render_event) {
+            case 'render':
+                this.dispatchEvent({ type: 'render' });
+                retVal = true;
+                break;
+            case 'rendercomplete':
+                this.dispatchEvent({ type: 'rendercomplete' });
+                retVal = true;
+                break;
+        }
+
+        return retVal;
+    }
+
     /**
     * Re-draw the layer panel to represent the current state of the layers.
     */
     renderPanel() {
-        LayerSwitcher.renderPanel(this.getMap(), this.panel, {
-            groupSelectStyle: this.groupSelectStyle
-        });
+        var panel_event;
+        panel_event = this.panelEvent("render");
+        LayerSwitcher.renderPanel(this.getMap(), this.panel, panel_event, { groupSelectStyle: this.groupSelectStyle });
+        this.panelEvent("rendercomplete");
```

Comment:
Same here `this.dispatchEvent({ type: 'rendercomplete' });` feels sufficient

Reply:
Yep, I thought all the panelEvent() method was necessary not to dispatch too many events, but indeed it is working fine without. Removed.

Reply:
Agree